libs/coin-framework/src/config.ts

```diff
@@ -30,12 +31,12 @@ function buildCoinConfig<T extends CurrencyConfig>() {
     coinConfig = config;
   };
 
-  const getCoinConfig = (): T => {
+  const getCoinConfig = (currency?: CryptoCurrency): T => {
     if (!coinConfig) {
       throw new MissingCoinConfig();
     }
 
```

Comment:
question : what if the currency parameter is undefined ?

Reply:
It's actually undefined for some chains, only Cosmos uses the currency to distinguish configs. 
https://github.com/LedgerHQ/ledger-live/blob/a0bb74b8f3704ab9d5567c9d14c16cab9e0872f7/libs/ledger-live-common/src/families/cosmos/setup.ts#L28


In xrp for instance, the method called doesn't expect any currency. https://github.com/LedgerHQ/ledger-live/blob/b14d37dc7bd4708950ecde4ace9b121c436935bc/libs/ledger-live-common/src/families/xrp/setup.ts#L19

It's optional as it allows some chains to only rely on one coin-config, and others (like Cosmos) to pass this parameter to get the proper config.